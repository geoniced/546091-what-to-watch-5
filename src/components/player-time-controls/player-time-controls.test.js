import React from "react";
import renderer from "react-test-renderer";
import PlayerTimeControls from "./player-time-controls";

describe(`PlayerTimeControls render`, () => {
  it(`renders PlayerTimeControls component with runtime eq 100 and currentTimeSeconds eq 3000`, () => {
    const tree = renderer
      .create(
          <PlayerTimeControls
            runtime={100}
            currentTimeSeconds={3000}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders PlayerTimeControls component with runtime eq 100 and currentTime eq 0`, () => {
    const tree = renderer
      .create(
          <PlayerTimeControls
            runtime={100}
            currentTimeSeconds={100}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders PlayerTimeControls component with runtime eq 100 and currentTime eq 6000`, () => {
    const tree = renderer
      .create(
          <PlayerTimeControls
            runtime={100}
            currentTimeSeconds={6000}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

