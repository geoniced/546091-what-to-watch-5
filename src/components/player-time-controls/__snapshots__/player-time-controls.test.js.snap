// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerTimeControls render renders PlayerTimeControls component with runtime eq 100 and currentTime eq 0 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={2}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "2%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    01:38:20
  </div>
</div>
`;

exports[`PlayerTimeControls render renders PlayerTimeControls component with runtime eq 100 and currentTime eq 6000 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={100}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    00:00:00
  </div>
</div>
`;

exports[`PlayerTimeControls render renders PlayerTimeControls component with runtime eq 100 and currentTimeSeconds eq 3000 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={50}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    00:50:00
  </div>
</div>
`;
